### PR TITLE
Add C petstore sample to Travis CI

### PR DESCRIPTION
**Pull request by wing328**
### PR checklist

- [ ] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [ ] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh` and `./bin/security/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`.
- [ ] Filed the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `3.3.x`, `4.0.x`. Default: `master`.
- [ ] Copied the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### Description of the PR

Test C petstore samples in Travis CI

### CODE REVIEW

File: samples/client/petstore/c/build-and-test.bash [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+sudo make install
+
+cd ..
+
+cmake .
+
+make 
+
+./unit-PetAPI
```
Comment (PowerOfCreation) [inline samples/client/petstore/c/build-and-test.bash]: This only executes one unit-test, change this to ctest and it should execute all unit-tests.

Comment (wing328) [reply samples/client/petstore/c/build-and-test.bash]: @PowerOfCreation thanks. I'll give it a try tomorrow.

**Comment by wing328** [comment]
@PowerOfCreation about using `ctest`, there are a few failed tests so we can't use it yet.
```
 6/14 Test  #6: unit-test-PetAPI .................***Exception: Other  0.19 sec
      Start  7: unit-test-StoreAPI
 7/14 Test  #7: unit-test-StoreAPI ...............***Exception: Other  0.13 sec
      Start  8: unit-test-UserAPI
 8/14 Test  #8: unit-test-UserAPI ................***Exception: Other  0.13 sec
      Start  9: unit-test-api-response
 9/14 Test  #9: unit-test-api-response ...........   Passed    0.00 sec
      Start 10: unit-test-category
10/14 Test #10: unit-test-category ...............   Passed    0.00 sec
      Start 11: unit-test-order
11/14 Test #11: unit-test-order ..................   Passed    0.00 sec
      Start 12: unit-test-pet
12/14 Test #12: unit-test-pet ....................   Passed    0.00 sec
      Start 13: unit-test-tag
13/14 Test #13: unit-test-tag ....................   Passed    0.00 sec
      Start 14: unit-test-user
14/14 Test #14: unit-test-user ...................   Passed    0.00 sec
79% tests passed, 3 tests failed out of 14
Total Test time (real) =   1.13 sec
The following tests FAILED:
	  6 - unit-test-PetAPI (OTHER_FAULT)
	  7 - unit-test-StoreAPI (OTHER_FAULT)
	  8 - unit-test-UserAPI (OTHER_FAULT)
```

cc @zhemant

**Comment by zhemant** [comment]
Whichever test are starting with unit-test-* are generated from a single file. they all are same. We have to exclude them. As of now can we use only file which are unit-*? As these file are manual test written for PetStore example.

**Comment by PowerOfCreation** [comment]
@zhemant @wing328 The unit-tests are not the same and they are working as expected and should definitely not be exlcluded. There are three tests that are failling. This is what happens when you keep doing new changes without executing the unit-tests to see if the changes are actually working as expected.

File: samples/client/petstore/c/build-and-test.bash [hunk 2]
```diff
@@ -22,4 +22,4 @@ cmake .
 
 make 
 
-ctest
+ctest -j <threads>
```
Comment (PowerOfCreation) [inline samples/client/petstore/c/build-and-test.bash]: Sorry if I was unclear: With <threads> I meant that you need to specifiy the amount of threads you want to use. I don't know how many cores your CI has, that is why I wrote it like that.
e.g.: $ ctest -j 4

**Comment by wing328** [comment]
@PowerOfCreation We'll not exclude the tests. Just that it's failing at the moment and we'll switch to `ctest` after these tests are fixed.

File: samples/client/petstore/c/build-and-test.bash (hunk 2 above)
Comment (wing328) [reply samples/client/petstore/c/build-and-test.bash]: Likely just one core so we should just go with `ctest` without the -j argument.

**Comment by zhemant** [comment]
Just to be clear files in "unit-test" folder are generated as unit-test-PetAPI, where as files in "unit-tests" folder are generated as unit-PetAPI. we dont need to run file which are in "unit-test" folder. according to logs seen above files in "unit-test" folder are running which are not the actual test.

**Comment by PowerOfCreation** [comment]
@zhemant Not sure what you mean. All unit-tests needs to be run and not just some. When you don't run the failing ones then you don't need to have unit-tests.

**Comment by zhemant** [comment]
https://github.com/OpenAPITools/openapi-generator/tree/c_generator/samples/client/petstore/c/unit-test 
Tests in above folder are not correct hence we don't need to run these. after compiling files in this folder they are named as unit-test-(mode/api name);


https://github.com/OpenAPITools/openapi-generator/tree/c_generator/samples/client/petstore/c/unit-tests
Test in this folder are correct. We only need to run these. After compiling files in this folder they are named as unit-(model/api name)

**Comment by wing328** [comment]
Closed via https://github.com/OpenAPITools/openapi-generator/pull/1136

**Pull request closed without merging**